scvi/model/_scvi.py

```diff
@@ -188,4 +203,72 @@ def setup_anndata(
             fields=anndata_fields, setup_method_args=setup_method_args
         )
         adata_manager.register_fields(adata, **kwargs)
+        # register new fields for latent mode if needed
+        latent_mode = _get_latent_adata_type(adata)
+        if latent_mode is not None:
+            adata_manager.register_new_fields(cls._get_latent_fields(latent_mode))
         cls.register_manager(adata_manager)
+
+    def _get_latent_adata_from_adata(
+        self,
+        mode: LatentDataType,
+        use_latent_key: str = "X_latent",
+        use_latent_qzm_key: str = "X_latent_qzm",
+        use_latent_qzv_key: str = "X_latent_qzv",
+    ) -> AnnData:
+        all_zeros = csr_matrix(self.adata.X.shape, dtype=np.int8).toarray()
+        bdata = AnnData(
+            X=all_zeros.copy(),
+            layers={layer: all_zeros.copy() for layer in self.adata.layers},
+        )
+        bdata.obs = self.adata.obs
+        bdata.obsm = self.adata.obsm
+        bdata.obsp = self.adata.obsp
+        bdata.uns = self.adata.uns
+        bdata.var_names = self.adata.var_names
+        # add latent related metadata
+        if mode == "sampled":
+            bdata.obsm[_SCVI_LATENT_SAMPLES] = self.adata.obsm[use_latent_key]
+        elif mode == "dist":
+            bdata.obsm[_SCVI_LATENT_QZM] = self.adata.obsm[use_latent_qzm_key]
+            bdata.obsm[_SCVI_LATENT_QZV] = self.adata.obsm[use_latent_qzv_key]
+        else:
+            raise ValueError(f"Unknown latent mode: {mode}")
+        bdata.uns[_ADATA_LATENT] = mode
+        return bdata
+
+    @staticmethod
+    def _get_latent_fields(mode: LatentDataType):
+        if mode == "sampled":
+            latent_fields = [
+                ObsmField(
+                    REGISTRY_KEYS.LATENT_SAMPLES_KEY,
+                    _SCVI_LATENT_SAMPLES,
+                ),
+            ]
+        elif mode == "dist":
+            latent_fields = [
+                ObsmField(
+                    REGISTRY_KEYS.LATENT_QZM_KEY,
+                    _SCVI_LATENT_QZM,
+                ),
+                ObsmField(
+                    REGISTRY_KEYS.LATENT_QZV_KEY,
+                    _SCVI_LATENT_QZV,
+                ),
+            ]
+        else:
+            raise ValueError(f"Unknown latent mode: {mode}")
+        return latent_fields
+
+    def to_latent_mode(
+        self,
+        mode: LatentDataType,
+        use_latent_key: str = "X_latent",
+        use_latent_qzm_key: str = "X_latent_qzm",
+        use_latent_qzv_key: str = "X_latent_qzv",
+    ):
```

Comment:
we should add the `latent_mode` to setup_anndata, at the moment I'm not seeing how load works properly (though I see it's tested). How do the latent fields get added when setup anndata gets run during the load method?

Reply:
another benefit of this is we can always use the manager to check if an anndata was used for latent mode.

Reply:
Can we create another abstract class that augments `BaseModelClass` with some functionality that one would need to add for latent mode?